### PR TITLE
Implement function / function expression config

### PR DESCRIPTION
**Pull request by eddeee888**
This PR adds a new config option that allows people to choose whether they want the component be written as function or function expression.

- Add config option to generate either function or function expression.
- Use added option to generate the JS/TS templates

TODO: 
[ ] Test - how does it work? Pls halp 💀

### CODE REVIEW

File: src/extension.ts [hunk 1]
```diff
@@ -16,10 +16,11 @@ export function activate(context: vscode.ExtensionContext) {
             const config = vscode.workspace.getConfiguration('createReactComponent');
             const isTypescript = config.get('language') === 'typescript';
 
-            const options = {
+            const options: TemplateOptions = {
                 name: componentName,
                 testLibrary: <TestLibrary>config.get('testingLibrary'),
                 cleanup: <boolean>config.get('testingLibrary.cleanup'),
+                functionType: <FunctionType>config.get('functionType')
```
Comment (yeet-bix) [inline src/extension.ts]: Some changes came in just before you created this PR. I have changed the type casts here to use the `as` syntax. You'll have to rebase 😬

File: src/template/javascriptTemplate.ts [hunk 2]
```diff
@@ -1,8 +1,9 @@
-import TemplateOptions, { TestLibrary } from './templateOptions';
+import TemplateOptions, { TestLibrary, FunctionType } from './templateOptions';
 
-const javascriptComponentTemplate = ({ name }: TemplateOptions) => `import React from 'react';
+const javascriptComponentTemplate = ({ name, functionType }: TemplateOptions) => `import React from 'react';
 
-function ${name}({}) {
+${functionType === FunctionType.Function && `function ${name}({}) {`}
+${functionType === FunctionType.Expression && `const ${name}: React.FunctionComponent<Props> = ({ }) => {`}
```
Comment (yeet-bix) [inline src/template/javascriptTemplate.ts]: Accidentally have types here in the javascript template

```suggestion
${functionType === FunctionType.Expression && `const ${name} = ({ }) => {`}
```

**Review by yeet-bix — CHANGES_REQUESTED** [review]
Just need to rebase but looks good! 👍 

FYI:
* `npm run test:unit`
* `npm run test:integration`

Don't worry about tests. 
The current test suite uses [sinon](https://sinonjs.org/) to mock the vscode window. I have created an issue here to refactor the tests: #7 and will add test coverage for this as part of the refactor 😄

**Comment by yeet-bix** [comment]
Could you add some docs for the setting in the README as well? 📝

**Comment by eddeee888** [comment]
I can't seem to run integration test in my machine.. The test window keeps crashing. Please test this before merging 😬

**Comment by yeet-bix** [comment]
The current version outputs:
```typescript
import React from 'react';

export interface MyComponentProps {}

function MyComponent({ }: MyComponentProps) {
false
    return <>MyComponent</>
};

export default MyComponent;
```

I think we will need something more explicit like a ternary:
```typescript
${functionType === FunctionType.Function ? `function ${name}({}) {` : ``}
${functionType === FunctionType.Expression ? `const ${name} = ({}) => {` : ``}
```


To manually test, you can either:

1. Press F5 and run the extension in a debug window
2. Run `vsce package` which will package the extension locally. You can then install the extension locally.

**Comment by eddeee888** [comment]
Thanks! I used "Extension Tests" instead of "Run extension" in debug mode which caused the test editor to open then close straight away 🤦‍♂ 

I have updated the code as well

**Review by yeet-bix — APPROVED** [review]
:shipit:

**Pull request merged**